cpp/src/io/orc/orc.cpp

```diff
@@ -356,53 +356,42 @@ size_t ProtobufWriter::write(const Metadata& s)
   return w.value();
 }
 
-OrcDecompressor::OrcDecompressor(CompressionKind kind, uint32_t blockSize)
-  : m_kind(kind), m_blockSize(blockSize)
-{
-  if (kind != NONE) {
-    int stream_type = IO_UNCOMP_STREAM_TYPE_INFER;  // Will be treated as invalid
-    switch (kind) {
-      case NONE: break;
-      case ZLIB:
-        stream_type    = IO_UNCOMP_STREAM_TYPE_INFLATE;
-        m_log2MaxRatio = 11;  // < 2048:1
-        break;
-      case SNAPPY:
-        stream_type    = IO_UNCOMP_STREAM_TYPE_SNAPPY;
-        m_log2MaxRatio = 5;  // < 32:1
-        break;
-      case LZO: stream_type = IO_UNCOMP_STREAM_TYPE_LZO; break;
-      case LZ4: stream_type = IO_UNCOMP_STREAM_TYPE_LZ4; break;
-      case ZSTD: stream_type = IO_UNCOMP_STREAM_TYPE_ZSTD; break;
-    }
-    m_decompressor = HostDecompressor::Create(stream_type);
-  } else {
-    m_log2MaxRatio = 0;
+OrcDecompressor::OrcDecompressor(CompressionKind kind, uint32_t blockSize) : m_blockSize(blockSize)
+{
+  switch (kind) {
+    case NONE:
+      _compression   = compression_type::NONE;
+      m_log2MaxRatio = 0;
+      break;
+    case ZLIB:
+      _compression   = compression_type::ZLIB;
+      m_log2MaxRatio = 11;  // < 2048:1
+      break;
+    case SNAPPY:
+      _compression   = compression_type::SNAPPY;
+      m_log2MaxRatio = 5;  // < 32:1
+      break;
+    case LZO: _compression = compression_type::LZO; break;
+    case LZ4: _compression = compression_type::LZ4; break;
+    case ZSTD: _compression = compression_type::ZSTD; break;
+    default: CUDF_FAIL("Invalid compression type");
   }
 }
 
-/**
- * @brief ORC block decompression
- *
- * @param[in] srcBytes compressed data
- * @param[in] srcLen length of compressed data
- * @param[out] dstLen length of uncompressed data
- *
- * @returns pointer to uncompressed data, nullptr if error
- */
-const uint8_t* OrcDecompressor::Decompress(const uint8_t* srcBytes, size_t srcLen, size_t* dstLen)
+host_span<uint8_t const> OrcDecompressor::decompress_blocks(host_span<uint8_t const> src)
 {
   // If uncompressed, just pass-through the input
-  if (m_kind == NONE) {
-    *dstLen = srcLen;
-    return srcBytes;
-  }
+  if (src.empty() or _compression == compression_type::NONE) { return src; }
+
+  constexpr int header_size = 3;
```

Comment:
Special thanks for this extraction of a magic value. Though it appears this is always added to `size_t` types -- maybe it should be a `size_t` as well.
```suggestion
  constexpr size_t header_size = 3;
```